contracts/src/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -96,17 +89,30 @@ contract SubnetActorCheckpointingFacet is SubnetActorModifiers, ReentrancyGuard,
             revert MaxMsgsPerBatchExceeded();
         }
 
-        // if the bottom up messages' length is max, we consider that epoch valid
-        if (checkpoint.msgs.length == s.maxMsgsPerBottomUpBatch) {
+        uint256 lastBottomUpCheckpointHeight = s.lastBottomUpCheckpointHeight;
+        uint256 bottomUpCheckPeriod = s.bottomUpCheckPeriod;
+
+        // cannot submit past bottom up checkpoint
+        if (checkpoint.blockHeight <= lastBottomUpCheckpointHeight) {
+            revert BottomUpCheckpointAlreadySubmitted();
+        }
+
+        uint256 nextCheckpointHeight = LibGateway.getNextEpoch(lastBottomUpCheckpointHeight, bottomUpCheckPeriod);
```

Comment:
ok so if I understand this right, then in case we had an early submission due to reaching msg size limit then we still continue to do checkpoints at multiples of `bottomUpCheckPeriod` after the early submission?

So for example if both `bottomUpCheckPeriod` and maxMsgsPerBottomUpBatch  is 10, and we had the following checkpoints and msg sizes:

chk 10 and 2 msg
chk 20 and 3 msg
chk 25 and 10msg -> here we had 10 msg already at height 25 causing early submission
chk 30 and 2 msg 
...

I s this correctly understood? I may have this wrong though as the screenshot that raul shares does not show that this is the case :S

Reply:
I thought the above screenshot is aligned with what you described? Just the above screenshot has checkpoint period 60. Am I missing sth?